pkgs/by-name/gi/gitstatus/package.nix

```diff
@@ -72,7 +80,9 @@ stdenv.mkDerivation rec {
     git
     zsh
   ];
-  doInstallCheck = true;
+  # On darwin:
+  # "zsh compinit: insecure directories, run compaudit for list."
+  doInstallCheck = stdenv.hostPlatform.isLinux;
```

Comment:
is this something specific to your machine? it's been fine on hydra and I also haven't seen this failure.
https://hydra.nixos.org/job/nixpkgs/trunk/gitstatus.aarch64-darwin

FWIW unvendoring fixes the clang build -- older zlibs don't compile with clang-19 and darwin

Reply:
maybe the something specific in your `/etc/zshrc` or similar is causing issues? maybe something like this

<details><summary>zsh -d</summary>

```diff
diff --git a/pkgs/by-name/gi/gitstatus/package.nix b/pkgs/by-name/gi/gitstatus/package.nix
index 04afd67b7c21..d0eab16e0dec 100644
--- a/pkgs/by-name/gi/gitstatus/package.nix
+++ b/pkgs/by-name/gi/gitstatus/package.nix
@@ -111,7 +111,7 @@ stdenv.mkDerivation rec {
     # run zsh as a full-fledged independent process and then wait for it to
     # exit. (The "exit" statements in the zshrc ensure that zsh will exit
     # almost immediately after starting.)
-    ZDOTDIR=. zsh -i &
+    ZDOTDIR=. zsh -d -i &
     wait $!
   '';
 
```
</details>

and thanks for taking up this PR -- I don't use this package but have seen enough of these vendored zlib issues from working on the llvm-19 update.

Reply:
yup. with sandbox disabled `/etc/zshenv` will get sourced
```
$ nix-build -E 'with import <nixpkgs>{}; runCommand "test" { nativeBuildInputs = [ zsh ]; } "zsh -o SOURCE_TRACE -c exit; exit 1"'
this derivation will be built:
  /nix/store/0aibdhalqv7apnhni3lfj0l8lq6a7l5y-test.drv
building '/nix/store/0aibdhalqv7apnhni3lfj0l8lq6a7l5y-test.drv'...
+/nix/store/v4l4c5spjykdl678rd7l69sxwxk09r5n-zsh-5.9/etc/zshenv:1> <sourcetrace>
+/etc/zshenv:1> <sourcetrace>
```
so perhaps you have a `zshenv` which is causing the issue. unfortunately, setting `-d` or `-f` doesn't prevent `zshenv` from getting sourced. anyway, I don't think this test should get disabled as it should run on hydra.

Reply:
> so perhaps you have a `zshenv` or `zshrc` which is causing the issue? I don't think this test should get disabled as it should run on hydra.

Maybe, but your original PR didn't finish evaluating for aarch64-Darwin and did not (yet) get build.

Can you point me to the hydra url which did build your PR?

Reply:
> > so perhaps you have a `zshenv` or `zshrc` which is causing the issue? I don't think this test should get disabled as it should run on hydra.
> 
> Maybe, but your original PR didn't finish evaluating for aarch64-Darwin and did not (yet) get build.

I built it and it failed for you. so there is either an issue with your machine or mine.

> Can you point me to the hydra url which did build your PR?

hydra has not built my PR but it build fine before the llvm-19 merge. please try to diagnose why your machine is failing before assuming that all machines fail. one thing to try is to see if you have the same problem when building before staging-next merged when hydra was successful.

additionally, it seems quite unlikely that dynamically linking to nixpkgs zlib rather than statically using the vendored zlib is causing your issues -- so the onus is on you to show that it is.

thanks.

Reply:
> maybe the something specific in your `/etc/zshrc` or similar is causing issues? maybe something like this to disable global rcs if you're not running in the sandbox(?).
> zsh -d
> 
> ```diff
> diff --git a/pkgs/by-name/gi/gitstatus/package.nix b/pkgs/by-name/gi/gitstatus/package.nix
> index 04afd67b7c21..d0eab16e0dec 100644
> --- a/pkgs/by-name/gi/gitstatus/package.nix
> +++ b/pkgs/by-name/gi/gitstatus/package.nix
> @@ -111,7 +111,7 @@ stdenv.mkDerivation rec {
>      # run zsh as a full-fledged independent process and then wait for it to
>      # exit. (The "exit" statements in the zshrc ensure that zsh will exit
>      # almost immediately after starting.)
> -    ZDOTDIR=. zsh -i &
> +    ZDOTDIR=. zsh -d -i &
>      wait $!
>    '';
>  
> ```
> 

this works, thanks!

> and thanks for taking up this PR -- I don't use this package but have seen enough of these vendored zlib issues from working on the llvm-19 update.

Sure, thanks for your patch!

Reply:
> I built it and it failed for you. so there is either an issue with your machine or mine.

As adding the `-d` flag to `zsh` fixes the build on my mac I suspect its a sandboxing issue. So I guess the issue was my machine ;-)

As adding `-d` flag doesn't cause an issue with the other builds and it fixes issues with sandboxed environments, I'd leave it